pkg/drivers/kic/oci/volumes.go

```diff
@@ -151,9 +151,9 @@ func createVolume(ociBin string, profile string, nodeName string) error {
 	return nil
 }
 
-// prepareVolume will copy the initial content of the mount point by starting a container to check the expected content
-func prepareVolume(ociBin string, imageName string, nodeName string) error {
-	cmdArgs := []string{"run", "--rm", "--entrypoint", "/usr/bin/test"}
+// prepareVolumeSideCar will copy the initial content of the mount point by starting a temporary container to check the expected content
+func prepareVolumeSideCar(ociBin string, imageName string, nodeName string) error {
```

Comment:
as part of this we should ensure the side car gets deleted as part of the clean up process
minikube delete

Reply:
https://github.com/medyagh/minikube/blob/d1758a355786707670e17f274553ee619fbb0b31/cmd/minikube/cmd/delete.go#L99


https://github.com/medyagh/minikube/blob/d1758a355786707670e17f274553ee619fbb0b31/cmd/minikube/cmd/delete.go#L155

and also 	deletePossibleKicLeftOver
https://github.com/medyagh/minikube/blob/d1758a355786707670e17f274553ee619fbb0b31/cmd/minikube/cmd/delete.go#L189

Reply:
Thank you for your point out!
This is important for the guarantee of deletion.

Reply:
I added the CreatedByLabelKey for preload-side car.

Reply:
@govargo can u verify if a side container is creatred and not deleted (remove --rm for testing)
the minikube delete would clean it up?

Reply:
I tested in my local with removing --rm. Then preload-side-car container kept living after `minikube delete`.

I investigated why the container kept living. And I noticed that `minikube delete` requieres ProfileLabelKey.
https://github.com/kubernetes/minikube/blob/40ac52ec4f35a3e8bc8017fb366c38899d78a974/cmd/minikube/cmd/delete.go#L249-L261

Reply:
I updated to include `ProfileLabelKey`.
I tested `minikube delete` again with removing --rm. And I confirmed that `minikube delete` deleted all containers including preload-side-car container.